src/objectExplorer/objectExplorerService.ts

```diff
@@ -65,21 +65,22 @@ export class ObjectExplorerService {
                 let nodeConnection = this._sessionIdToConnectionCredentialsMap.get(result.sessionId);
                 for (let connection of savedConnections) {
                     if (Utils.isSameConnection(connection.connectionCreds, nodeConnection)) {
-                        nodeLabel = connection.label;
-                        break;
+                        // if it's not the defaul label
+                        if (connection.label !== connection.connectionCreds.server) {
+                            nodeLabel = connection.label;
+                            break;
+                        }
                     }
                 }
                 // set connection and other things
                 if (self._currentNode && (self._currentNode.sessionId === result.sessionId)) {
-                    nodeLabel = nodeLabel === result.rootNode.nodePath ?
-                    self.createNodeLabel(self._currentNode.connectionCredentials) : nodeLabel;
+                    nodeLabel = !nodeLabel ? self.createNodeLabel(self._currentNode.connectionCredentials) : nodeLabel;
```

Comment:
nit: but you can shorten this slightly by doing 
`nodeLabel = nodeLabel || self.createNodeLabel(self._currentNode.connectionCredentials);`